R/sits_kohonen.R

```diff
@@ -1,92 +1,770 @@
-#' @title Cluster a set of satellite image time series using SOM
+#' @title Clustering a set of satellite image time series using SOM
 #' @name sits_kohonen
-#' @author Gilberto Camara, \email{gilberto.camara@@inpe.br}
-#' @author Rolf Simoes, \email{rolf.simoes@@inpe.br}
 #' @author Lorena Alves, \email{lorena.santos@@inpe.br}
 #'
-#' @description This function uses package "kohonen" (self-organized maps) to find clusters in
-#' satellite image time series.
+#' @description This function uses package self-organized maps to find clusters in
+#' satellite image time series to cluster samples.
 #' @references `kohonen` package (https://CRAN.R-project.org/package=kohonen)
 #'
-#' @param data.tb        A tibble with time series to be clustered.
-#' @param bands          Bands to be clustered.
-#' @param grid_xdim      X dimension of the SOM grid (default = 5).
-#' @param grid_ydim      Y dimension of the SOM grid.
-#' @param rlen           Number of times the complete data set will be presented to the SOM grid.
-#' @param alpha          Learning rate, a vector of two numbers indicating the amount of change.
+#' @param data.tb        a tibble with samples to be clustered
+#' @param time_series    time series extracted from tibble data.
+#' @param bands          bands to be clustered.
+#' @param grid_xdim      x dimension of the SOM grid (default = 5)
+#' @param grid_ydim      y dimension of the SOM grid
+#' @param rlen           number of times the complete data set will be presented to the SOM grid
+#' @param alpha          learning rate, a vector of two numbers indicating the amount of change.
 #'                       Default is to decline linearly from 0.05 to 0.01 over rlen updates.
-#' @param  ...           Additional parameters to be passed to kohonen::supersom function.
-#' @return A full kohonen object for data analysis.
-#' @examples
-#' \donttest{
-#' # Retrieve a time series to be clusterized
-#' data("samples_MT_9classes")
-#' # compute a kohonen object
-#' koh_obj <- sits_kohonen(samples_MT_9classes)
-#' # generate a dataset
-#' cluster.tb <- sits_kohonen_cluster(samples_MT_9classes, koh_obj)
-#' # plot the result
-#' sits_cluster_frequency(cluster.tb)
-#' }
+#' @param  ...           additional parameters to be passed to kohonen::supersom function
+#' @return clusters.tb   a tibble with the clusters time series or cluster' members time series according to return_member parameter.
+#' If return_members are FALSE, the returning tibble will contain a new collumn called `n_members` informing how many members has each cluster.
 #' @export
-sits_kohonen <- function (data.tb, bands = NULL, grid_xdim = 5, grid_ydim = 5, rlen = 100,
-                          alpha = c(0.05, 0.01), ...) {
+sits_kohonen <- function (data.tb, time_series, bands = NULL, grid_xdim = 25, grid_ydim = 25, rlen = 100,dist.fcts="euclidean",
+                          alpha = 1,decay.fcts="exponential",neighbourhood.fct ="bubble", ...) {
+
```

Comment:
Please define a version name in lorenalves/kohonen that does not match any version of kohonen package in CRAN, and check this version here. If the required version is not the installed version, stop with an error telling the user to install lorenalves/kohonen.